src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -212,6 +357,11 @@ def wrapper(*args, **kwargs):
                     #  Handle retriable exception, please refer to
                     #  https://platform.openai.com/docs/guides/error-codes/api-errors
                     print(f"Exception occurs: {type(e).__name__}: {str(e)}", file=sys.stderr)
+                    if isinstance(e, BadRequestError) and \
+                            ("extra fields not permitted" in str(e).lower()
+                             or "none is not an allowed value" in str(e).lower()):
```

Comment:
should not meet `none is not an allowed value` after we do not pass any None value to openai request.